go/vt/discovery/tablet_stats_cache.go

```diff
@@ -357,21 +361,29 @@ func (tc *TabletStatsCache) updateAggregateMap(keyspace, shard string, tabletTyp
 	newAgg := MakeAggregateMap(stats)
 	e.aggregates = newAgg
 
-	// And broadcast the change in the background.
-	go func() {
+	// And broadcast the change in the background, if we need to.
+	if !tc.tsm.HasSubscribers() {
+		// Shortcut: no subscriber, we can be done.
+		return
+	}
+	var ter int64
+	if len(stats) > 0 {
+		ter = stats[0].TabletExternallyReparentedTimestamp
+	}
+	diffs := MakeAggregateMapDiff(keyspace, shard, tabletType, ter, oldAgg, newAgg)
+	tc.aggregatesChan <- diffs
+}
+
+// broadcastAggregateStats is called in the background to send aggregate stats
+// in the right order to our subscribers.
+func (tc *TabletStatsCache) broadcastAggregateStats() {
+	for diffs := range tc.aggregatesChan {
```

Comment:
Is it okay that you're never going to close this channel?

Will the binaries shut down cleanly anyway?

Reply:
They do, yes. There is no shutdown at this point for this code, and we have other channels in other places that are never closed.